### DISCUSSION TITLE
Help needed with import results

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#1234] [General]
I feel a bit silly asking this question here, but I am just not seeing straight here...
A while ago I spoke to Josh about an change I wanted to make in the details page for meetings.
Before the change, the column next to the map would show, in the bottom, the group details.
I wanted to change that and I managed to do so.

I am now preparing a handover to my successor and I notice that an uploaded meeting gives me different results on the live website and the test website.

I have been staring at the functions.php file in both websites and I can find no difference. But the import from the same document uploads differently. I have checked my old emails but I cannot for the life of me find out what I did.
I know we solved it because the live website is doing what I intended.

I hope someone of the dev team remembers how to do this so I can repeat it on the test website.

I am attaching two pics. Although they are of different groups, the principle is the same:
The 'Live' shows what I want on the test website: I do not wish to repeat the group information in the bottom section, but show a generic text for newcomers:
![Live](https://github.com/code4recovery/12-step-meeting-list/assets/59859095/83448cae-9d34-41df-8d9b-ad081ad8405c)

TEST shows what I do NOT want: the repeat of the group information
![Test](https://github.com/code4recovery/12-step-meeting-list/assets/59859095/edf3ba6b-4dd0-4fd2-a23c-f334389c10b2)

**Comment by joshreisner** [comment]
was this the fix? https://github.com/code4recovery/12-step-meeting-list/discussions/1195#discussioncomment-6786296

**Reply by joshreisner** [reply]
@OSAwebsite no, in the comment linked to above I showed you a way you could hide the district in the group section. The "District 32" in your second screenshot.

i don't know how you added generic text to the group instead of showing the information stored in the notes field, perhaps you made a local copy of single-meeting.php? 

here is another comment relevant to group notes in that thread: https://github.com/code4recovery/12-step-meeting-list/discussions/1195#discussioncomment-6786853

> things to know about groups: they are unique by name and are shared between meetings. so when editing a group that affects all meetings with that group. but this has an implication for importing: if you have several rows with the Group name One Day at a Time then it will only take the group notes / contacts / etc from ONE of those rows.

**Comment by OSAwebsite** [comment]
I checked, no, that's not it. That was about the headers of the filter
boxes in the list.
I know that this was really something simple, so simple that I cannot for
the life of me find it.
It is in the content of the group notes. The Newcomer info that is in there
in the .csv is accepted in the live site but not in Osatest.
I have doublechecked CSS as well. Is there another place in the
'Appearance' section of Wordpress that may do this?


-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
You could probably find it if you put something in group notes and then get
that to show instead of the repeated group information after import. I am
not sure that I would have to redo an import after finding out how we did
it...


-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
I read that too.

I will do an experiment. Delete all meetings, empty the trash and if there
are still data in regions and districts delete them too. Complete empty
slate. Then import the one meeting either the newcomer info to see if it
uploads and displays. If so, I think that the TSML import function, even if
I indicate to delete everything, might still hold on to group info in the
back.

I will do the deletes either the TSML import first: it is quicker than
using the bulk action. But the I do have to use them to empty the bin.

Let’s see where that gets me.

**Comment by OSAwebsite** [comment]
OK. Now I have a better understanding of what is happening. It is all due to the way the TSML code handles the difference between groups and meetings, versus how I do it in my database.
When I use the lightning fast 'delete all' option  while importing just one test meeting (just so the import runs), all meetings except the test meeting are deleted. That is great. The test meeting shows the 'NEWCOMER INFORMATION' in group notes as well.
I then delete the newly imported meeting. And then go to the thrash and click 'Empty trash'. Also nice and fast and much better than the bulk action delete-equivalent. Great too.
Then I check regions and districts, that are now empty.
When I now re-import the full set, and the 'NEWCOMER INFORMATION' is shown for every meeting. That is great!

So my cautious conclusion is that for this to work, I need to make sure manually that everything is deleted before the import.
I thought that that was what I had done for the OSA test refresh. But I must have missed something.

It is good to have clarity about this now.
The other thing I now also understand is this:

After checking in the front end the listings that have errors in their tags are all groups with multiple meetings.
All online meetings of groups that also have a  (separate) in-person meeting get the tag 'online and in-person' and the toggle in those online meetings has gone from No to Yes.

We have talked about this before. The automated assumptions the coding makes get in my way.  There is no option in the import to include that toggle. It would be awesome if there was. It might make the coding 'lighter' as well, maybe?
So, when I import a group with an in-person meeting AND a separate online meeting ( which will happen more and more now that we have global electronic meetings in Al-Anon that are accepted as a part of local districts and/or areas), it is the TSML code that takes the meeting format based on one of the two. Probably the in-person one, as that one is considered the default meeting format. But that will not be true for what is happening in reality. We will keep having online meetings run by the same group. And for those, I now have to manually correct the toggle. Would be nice if the toggle could be a straight forward Yes/No option that we can include in the importsheet.

But at least now I know what to look for when I do final checks after a completely new upload.

I hope this makes sense and maybe even is helpful information to the team.

As you probably know, another Webcoordinator will soon take over from me as I rotate out. So I may not be so involved with the plugin anymore. But I want to thank you and the others on the team for always so generously looking into the matters I brought to the table. You are all doing a terrific job and the way meetings are displayed is such a vast improvement from the pre-TSML era, that I am filled with gratitude. For newcomers this has made a huge difference and for us, who work with your plugin, it is a joy because it all has become so simple and for the most part straight forward.

My wishlist only has two major things:
-having the WSO ID (or equivalent thereof in different 12 Step Programs) in its own input field and included in a search in de filter box
-Having the in-person toggle as a separate value included in the importsheet, so that import results are more predictable.

Other than that: big round of applause!

With program love and appreciation,
